src/main/java/org/openrewrite/java/micronaut/ChangeAnnotationProcessorPath.java

```diff
@@ -112,51 +111,39 @@ private Xml.Tag maybeUpdateConfiguration(Xml.Tag configuration, ExecutionContext
 
             private Xml.Tag maybeUpdateAnnotationProcessorPaths(Xml.Tag annotationProcessorPaths, ExecutionContext ctx) {
                 return mapTagChildren(annotationProcessorPaths, childTag -> {
-                   if ("path".equals(childTag.getName()) && isPathMatch(childTag)) {
-                       Xml.Tag path = childTag;
-                       if (newGroupId != null) {
-                           path = changeChildTagValue(path, "groupId", newGroupId, ctx);
-                       }
-                       if (newArtifactId != null) {
-                           path = changeChildTagValue(path, "artifactId", newArtifactId, ctx);
-                       }
-                       if (newVersion != null) {
-                           path = changeChildTagValue(path, "version", newVersion, ctx);
-                       }
-                       if (exclusions == null) {
-                           path = filterTagChildren(path, child -> !("exclusions".equals(child.getName())));
-                       } else if (exclusions != null) {
-                           path = addExclusionsToPath(path, ctx);
-                       }
-                       childTag = path;
-                   }
-                   return childTag;
+                    if ("path".equals(childTag.getName()) && isPathMatch(childTag)) {
+                        Xml.Tag path = childTag;
+                        if (newGroupId != null && !newGroupId.equals(path.getChildValue("groupId").orElse(""))) {
+                            path = changeChildTagValue(path, "groupId", newGroupId, ctx);
+                        }
+                        if (newArtifactId != null && !newArtifactId.equals(path.getChildValue("artifactId").orElse(""))) {
+                            path = changeChildTagValue(path, "artifactId", newArtifactId, ctx);
+                        }
+                        if (newVersion != null) {
+                            String versionToUpdate = newVersion.startsWith("micronaut.") ? "${" + newVersion + "}" : newVersion;
+                            if (!versionToUpdate.equals(path.getChildValue("version").orElse(""))) {
+                                path = changeChildTagValue(path, "version",
+                                        versionToUpdate,
+                                        ctx);
+                            }
+                        }
+                        if (exclusions == null) {
+                            path = filterTagChildren(path, child -> !("exclusions".equals(child.getName())));
+                        } else {
```

Comment:
Any risk of this adding and empty `<exclusions></exclusions>` block that we might want to avoid?